src/Microsoft.Fx.Portability/IAssemblyFile.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+
+namespace Microsoft.Fx.Portability
+{
+    public interface IAssemblyFile
+    {
+        Stream OpenRead();
+
+        string Path { get; }
```

Comment:
Seems good (in that it's a more general interface). Do we know if this will work with our metadata reader? A quick perusal shows that we attempt to gather assembly information by loading from disk. Are more changes needed there to enable a non-file-based assembly to work?